source/bdDetect.py

```diff
@@ -221,51 +239,54 @@ def getDriversForPossibleBluetoothDevices() -> typing.Iterator[typing.Tuple[str,
 				)
 
 
+btDevsCacheT = Optional[List[Tuple[str, DeviceMatch]]]
+
+
 class _DeviceInfoFetcher(AutoPropertyObject):
 	"""Utility class that caches fetched info for available devices for the duration of one core pump cycle."""
 	cachePropertiesByDefault = True
 
 	def __init__(self):
-		self._btDevsLock = threading.Lock
-		self._btDevsCache: typing.Optional[typing.Tuple[str, DeviceMatch]] = None
+		self._btDevsLock = threading.Lock()
+		self._btDevsCache: Optional[List[Tuple[str, DeviceMatch]]] = None
 
 	#: Type info for auto property: _get_btDevsCache
-	btDevsCache: typing.Optional[typing.List[typing.Tuple[str, DeviceMatch]]]
+	btDevsCache: btDevsCacheT
 
-	def _get_btDevsCache(self):
-		with self._btDevsLock():
+	def _get_btDevsCache(self) -> btDevsCacheT:
+		with self._btDevsLock:
 			return self._btDevsCache
 
 	def _set_btDevsCache(
 			self,
-			cache: typing.Optional[typing.List[typing.Tuple[str, DeviceMatch]]]
+			cache: btDevsCacheT,
 	):
-		with self._btDevsLock():
+		with self._btDevsLock:
 			self._btDevsCache = cache

```

Comment:
We should also keep an internal copy, so create a new copy when given a cache

```suggestion
			self._btDevsCache = cache.copy()
```